### PR TITLE
fix(map): Removed NA text for challenge without time estimat

### LINKED ISSUE #16454: Change "N/A" to blank in map time estimates
Instead of showing "N/A" in the map for the interview prep and open source for good sections, we just shouldn't show a time estimate - it should be blank.

project_euler_problems__problem_7__10001st_prime___freecodecamp

### PR DESCRIPTION
**Pull request by lgrzybowski**
Closes #16454

#### Pre-Submission Checklist
- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change
- [X] Small bug fix (non-breaking change which fixes an issue)

#### Checklist:

- [X] Tested changes locally.
- [X] Addressed currently open issue https://github.com/freeCodeCamp/freeCodeCamp/issues/16454

Closes #16454

#### Description
I had run already implemented tests, but I think for react project it would be really nice to use jest with [snapshot testing](https://facebook.github.io/jest/docs/en/snapshot-testing.html). One more idea would be to add eslint check as part of `npm test` command

### CODE REVIEW

**Comment by vkWeb** [comment]
@lgrzybowski Next time please comment on the issue you're working on so that the other participants know that you're on it.

**Comment by 0x0936** [comment]
The if/else block added in Block.jsx produces a lot of duplicate code. Perhaps there's a more elegant way to inject the time span.

```js
{
  time
    ? <span className={ `${ns}-block-time` }>({ time })</span>
    : ''
}
```

**Comment by r1chard5mith** [comment]
Or you could just leave the time as N/A and test for it in the JSX and then it's a one line change to one file only.

`<span className={ `${ns}-block-time` }>{ time === "N/A" ? "" : "(" + time + ")" }</span>
`

**Comment by lgrzybowski** [comment]
@0x0936 This should looks better now.

File: common/app/Map/Block.jsx [hunk 1]
```diff
@@ -61,7 +61,11 @@ export class Block extends PureComponent {
         <span>
         { title }
         </span>
-        <span className={ `${ns}-block-time` }>({ time })</span>
+        {
+          time
+          ? <span className={ `${ns}-block-time` }>({ time })</span>
+          : ''
```
Comment (BerkeleyTrue) [inline common/app/Map/Block.jsx]: This can be reduced to a short circuit
```js
foo && <foo />
``

**Review by BerkeleyTrue — CHANGES_REQUESTED** [review]

**Comment by lgrzybowski** [comment]
@BerkeleyTrue Thx for your review this should be looking better now.

**Review by BerkeleyTrue — APPROVED** [review]

**Comment by BerkeleyTrue** [comment]
@lgrzybowski nice work!


Happy Coding

**Comment by QuincyLarson** [comment]
@lgrzybowski Awesome! Thank you for your contribution to the codebase!

**Pull request merged**